Error creating azurerm_hdinsight_kafka_cluster for HDInsight on Azure with Terraform

### Is there an existing issue for this?

- [x] I have searched the existing issues

### Community Note

<!--- Please keep this note for the community --->

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave comments along the lines of "+1", "me too" or "any updates", they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

<!--- Thank you for keeping this note for the community --->


### Terraform Version

1.11.4

### AzureRM Provider Version

4.27.0

### Affected Resource(s)/Data Source(s)

azurerm_4.27.0

### Terraform Configuration Files

```hcl
module definition:
Kafka:
=====
module "test-p00-kafka-cluster" {
  source                  = "git::repo/_git/terraform-azure-hdinsight-kafka?ref=v4.0.0"
  location                = var.location
  resource_group_name     = module.test-p00-kafka-rg.name
  enable_kafka_monitoring = false
  tags                    = var.env_tags

  kafka_cluster = {
    name                          = lower(""test-d00-hdi-kafka"")
    cluster_version               = "5.1"
    kafka_version                 = "3.2"
    tier                          = "Standard"
    tls_min_version               = "1.2"
    encryption_in_transit_enabled = true
    gateway_username              = var.kafka_gateway_username
    gateway_password              = var.kafka_gateway_password
  }

  kafka_roles = {
    vm_username = var.kafka_vm_username
    vm_password = var.kafka_vm_password

    head_node = {
      vm_size            = var.kafka_head_node_vm_size
      subnet_id          = data.terraform_remote_state.shared_platform_management.outputs.vnet_10_81_0_0_subnets[10]
      virtual_network_id = data.terraform_remote_state.shared_platform_management.outputs.vnet_10_81_0_0_id
      script_actions     = []
    }

    worker_node = {
      vm_size                  = var.kafka_worker_node_vm_size
      subnet_id                = data.terraform_remote_state.shared_platform_management.outputs.vnet_10_81_0_0_subnets[10]
      virtual_network_id       = data.terraform_remote_state.shared_platform_management.outputs.vnet_10_81_0_0_id
      target_instance_count    = 3
      number_of_disks_per_node = 2
      script_actions           = []
    }

    zookeeper_node = {
      vm_size            = var.kafka_zookeeper_node_vm_size
      subnet_id          = data.terraform_remote_state.shared_platform_management.outputs.vnet_10_81_0_0_subnets[10]
      virtual_network_id = data.terraform_remote_state.shared_platform_management.outputs.vnet_10_81_0_0_id
      script_actions     = []
    }

    kafka_management_node = null
  }

  kafka_storage_account = {
    storage_account_key  = module.test-p00-kafka-sa01.primary_account_key
    storage_container_id = module.test-p00-kafka-sa01.storage_containers["hdinsight"].id
    storage_resource_id  = module.test-p00-kafka-sa01.storage_account_id
  }

  depends_on = [
    module.test-p00-kafka-rg,
    module.test-p00-kafka-sa01
  ]
}

storage account:
=================
module "test-p00-kafka-sa01" {
  source               = "git::repo/_git/terraform-azure-storage-account?ref=v4.0.6"
  resource_group_name  = module.test-p00-kafka-rg.name
  storage_account_name = lower("testdp00hdi")
  location             = var.location
  tags                 = var.env_tags

  subnet_ids = [
    data.terraform_remote_state.shared_platform_management.outputs.vnet_10_81_0_0_subnets[10]
  ]

  containers = [
    {
      name        = "hdinsight"
      access_type = "private"
    },
  ]

  depends_on = [
    module.test-p00-kafka-rg
  ]
}
```

### Debug Output/Panic Output

```shell
Terraform plan:
  # module.emea-weu-d-shd-p00-kafka-cluster.azurerm_hdinsight_kafka_cluster.main will be created
  + resource "azurerm_hdinsight_kafka_cluster" "main" {
      + cluster_version               = "5.1"
      + encryption_in_transit_enabled = true
      + https_endpoint                = (known after apply)
      + id                            = (known after apply)
      + kafka_rest_proxy_endpoint     = (known after apply)
      + location                      = "westeurope"
      + name                          = "test-d00-hdi-kafka"
      + resource_group_name           = "TEST-D00-KAFKA"
      + ssh_endpoint                  = (known after apply)
      + tags                          = {
          + "subPurpose"     = "Development"
        }
      + tier                          = "Standard"
      + tls_min_version               = "1.2"

      + component_version {
          + kafka = "3.2"
        }

      + gateway {
          + password = (sensitive value)
          + username = "admin"
        }

      + roles {
          + head_node {
              + password           = (sensitive value)
              + subnet_id          = "/subscriptions/xxxxxx/resourceGroups/xxxxxx/providers/Microsoft.Network/virtualNetworks/xxxxxx/subnets/TEST-D00-KAFKA-SUBNET-10.81.10.0_24"
              + username           = "wspadmin"
              + virtual_network_id = "/subscriptions/xxxxxx/resourceGroups/xxxxxx/providers/Microsoft.Network/virtualNetworks/xxxxxx"
              + vm_size            = "Standard_D3_V2"
            }
          + worker_node {
              + number_of_disks_per_node = 2
              + password                 = (sensitive value)
              + subnet_id                = "/subscriptions/xxxxxx/resourceGroups/xxxxxx/providers/Microsoft.Network/virtualNetworks/xxxxxx/subnets/TEST-D00-KAFKA-SUBNET-10.81.10.0_24"
              + target_instance_count    = 3
              + username                 = "wspadmin"
              + virtual_network_id       = "/subscriptions/xxxxxx/resourceGroups/xxxxxx/providers/Microsoft.Network/virtualNetworks/xxxxxx"
              + vm_size                  = "Standard_D3_V2"
            }
          + zookeeper_node {
              + password           = (sensitive value)
              + subnet_id          = "/subscriptions/xxxxxx/resourceGroups/xxxxxx/providers/Microsoft.Network/virtualNetworks/xxxxxx/subnets/TEST-D00-KAFKA-SUBNET-10.81.10.0_24"
              + username           = "wspadmin"
              + virtual_network_id = "/subscriptions/xxxxxx/resourceGroups/xxxxxx/providers/Microsoft.Network/virtualNetworks/xxxxxx"
              + vm_size            = "Standard_D3_V2"
            }
        }

      + storage_account {
          + is_default           = true
          + storage_account_key  = (sensitive value)
          + storage_container_id = "/subscriptions/xxxxxx/resourceGroups/TEST-D00-KAFKA/providers/Microsoft.Storage/storageAccounts/edshdd00hdi/blobServices/default/containers/hdinsight"
          + storage_resource_id  = "/subscriptions/xxxxxx/resourceGroups/TEST-D00-KAFKA/providers/Microsoft.Storage/storageAccounts/edshdd00hdi"
        }
    }


Error:
    ╷
│ Error: creating Kafka HDInsight Cluster (Subscription: "xxxxxx"
│ Resource Group Name: "TEST-D00-KAFKA"
│ Cluster Name: "test-d00-hdi-kafka"): performing Create: unexpected status 400 (400 Bad Request) with response: {"code":"BadRequest","message":"Storage account 'name' in storage profile cannot be null or empty"}
│ 
│   with module.emea-weu-d-shd-p00-kafka-cluster.azurerm_hdinsight_kafka_cluster.main,
│   on .terraform/modules/emea-weu-d-shd-p00-kafka-cluster/main.tf line 30, in resource "azurerm_hdinsight_kafka_cluster" "main":
│   30: resource "azurerm_hdinsight_kafka_cluster" "main" {
│ 
╵

##[error]Script failed with exit code: 1
```

### Expected Behaviour

It should create the cluster but its showing error


Other Resources used:
 - Virtual Network
 - Dedicated subnet for HDinsights Kafka cluster
 - Service endpoints on the subnet ("Microsoft.Storage", "Microsoft.KeyVault", "Microsoft.Sql", "Microsoft.AzureCosmosDB", "Microsoft.CognitiveServices", "Microsoft.ServiceBus", "Microsoft.EventHub", "Microsoft.ContainerRegistry")

 -  NSG for HDInsights Cluster
![Image](https://github.com/user-attachments/assets/e9952b43-4ac9-48e1-985f-452e6b3ce1cb)


### Actual Behaviour

Giving error
│ Cluster Name: "test-d00-hdi-kafka"): performing Create: unexpected status 400 (400 Bad Request) with response: {"code":"BadRequest","message":"Storage account 'name' in storage profile cannot be null or empty"}


### Steps to Reproduce

_No response_

### Important Factoids

https://docs.azure.cn/en-us/hdinsight/kafka/apache-kafka-quickstart-resource-manager-template
https://docs.azure.cn/en-us/hdinsight/kafka/apache-kafka-quickstart-bicep?tabs=CLI

as per the above Azure decumentation, both of them passing storage account name

    storageProfile: {
      storageaccounts: [
        {
          name: replace(replace(concat(reference(storageAccount.id, '2021-08-01').primaryEndpoints.blob), 'https:', ''), '/', '')
          isDefault: true
          container: clusterName
          key: listKeys(storageAccount.id, '2021-08-01').keys[0].value
        }
      ]
    }

Are we sure that the terraform resource works without storage_account_name?

### References

https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/hdinsight_kafka_cluster.html
https://learn.microsoft.com/en-us/azure/hdinsight/hdinsight-management-ip-addresses
Similar Issue: https://github.com/hashicorp/terraform-provider-azurerm/issues/25125

--- a/website/docs/r/hdinsight_hadoop_cluster.html.markdown
+++ b/website/docs/r/hdinsight_hadoop_cluster.html.markdown
@@ -215,7 +215,7 @@ A `storage_account` block supports the following:
 
 * `storage_container_id` - (Required) The ID of the Storage Container. Changing this forces a new resource to be created.
 
--> **Note:** This can be obtained from the `id` of the `azurerm_storage_container` resource.
+-> **Note:** When the `azurerm_storage_container` resource is created with `storage_account_name`, this can be obtained from the `id` of the `azurerm_storage_container` resource. When the `azurerm_storage_container` resource is created with `storage_account_id`, please use `azurerm_storage_containers` data source to get the `data_plane_id` of the `azurerm_storage_container` resource for this field.
 
 * `storage_resource_id` - (Optional) The ID of the Storage Account. Changing this forces a new resource to be created.
 

--- a/website/docs/r/hdinsight_hbase_cluster.html.markdown
+++ b/website/docs/r/hdinsight_hbase_cluster.html.markdown
@@ -213,7 +213,7 @@ A `storage_account` block supports the following:
 
 * `storage_container_id` - (Required) The ID of the Storage Container. Changing this forces a new resource to be created.
 
--> **Note:** This can be obtained from the `id` of the `azurerm_storage_container` resource.
+-> **Note:** When the `azurerm_storage_container` resource is created with `storage_account_name`, this can be obtained from the `id` of the `azurerm_storage_container` resource. When the `azurerm_storage_container` resource is created with `storage_account_id`, please use `azurerm_storage_containers` data source to get the `data_plane_id` of the `azurerm_storage_container` resource for this field.
 
 * `storage_resource_id` - (Optional) The ID of the Storage Account. Changing this forces a new resource to be created.
 

--- a/website/docs/r/hdinsight_interactive_query_cluster.html.markdown
+++ b/website/docs/r/hdinsight_interactive_query_cluster.html.markdown
@@ -217,7 +217,7 @@ A `storage_account` block supports the following:
 
 * `storage_container_id` - (Required) The ID of the Storage Container. Changing this forces a new resource to be created.
 
--> **Note:** This can be obtained from the `id` of the `azurerm_storage_container` resource.
+-> **Note:** When the `azurerm_storage_container` resource is created with `storage_account_name`, this can be obtained from the `id` of the `azurerm_storage_container` resource. When the `azurerm_storage_container` resource is created with `storage_account_id`, please use `azurerm_storage_containers` data source to get the `data_plane_id` of the `azurerm_storage_container` resource for this field.
 
 * `storage_resource_id` - (Optional) The ID of the Storage Account. Changing this forces a new resource to be created.
 

--- a/website/docs/r/hdinsight_kafka_cluster.html.markdown
+++ b/website/docs/r/hdinsight_kafka_cluster.html.markdown
@@ -212,7 +212,7 @@ A `storage_account` block supports the following:
 
 * `storage_container_id` - (Required) The ID of the Storage Container. Changing this forces a new resource to be created.
 
--> **Note:** This can be obtained from the `id` of the `azurerm_storage_container` resource.
+-> **Note:** When the `azurerm_storage_container` resource is created with `storage_account_name`, this can be obtained from the `id` of the `azurerm_storage_container` resource. When the `azurerm_storage_container` resource is created with `storage_account_id`, please use `azurerm_storage_containers` data source to get the `data_plane_id` of the `azurerm_storage_container` resource for this field.
 
 * `storage_resource_id` - (Optional) The ID of the Storage Account. Changing this forces a new resource to be created.
 

--- a/website/docs/r/hdinsight_spark_cluster.html.markdown
+++ b/website/docs/r/hdinsight_spark_cluster.html.markdown
@@ -217,7 +217,7 @@ A `storage_account` block supports the following:
 
 * `storage_container_id` - (Required) The ID of the Storage Container. Changing this forces a new resource to be created.
 
--> **Note:** This can be obtained from the `id` of the `azurerm_storage_container` resource.
+-> **Note:** When the `azurerm_storage_container` resource is created with `storage_account_name`, this can be obtained from the `id` of the `azurerm_storage_container` resource. When the `azurerm_storage_container` resource is created with `storage_account_id`, please use `azurerm_storage_containers` data source to get the `data_plane_id` of the `azurerm_storage_container` resource for this field.
 
 * `storage_resource_id` - (Optional) The ID of the Storage Account. Changing this forces a new resource to be created.
 

